### PR TITLE
Add events data to GetTransaction rpc

### PR DESCRIPTION
**Pull request by psheth9**
More details here --> https://github.com/stellar/soroban-rpc/issues/16 

### Known limitations
NA

### CODE REVIEW

**Comment by stellarsaur** [comment]
Once ready for review, please close this PR and reopen in `soroban-rpc`. We could also merge this PR here when ready and cherry pick over to `soroban-rpc`, but it would be easier to directly merge this feature to `soroban-rpc`.

File: cmd/soroban-rpc/internal/transactions/transactions_test.go [hunk 1]
```diff
@@ -311,6 +347,21 @@ func TestIngestTransactions(t *testing.T) {
...
+	// Insert ledger 1
+	metaWithEvents := txMetaWithEvents(1, false)
+	require.NoError(t, store.IngestTransactions(metaWithEvents))
+	require.Len(t, store.transactions, 1)
+
+	// check events data
+	tx, ok, _ := store.GetTransaction(txHash(1, false))
+	require.True(t, ok)
+	require.NotNil(t, tx.Events)
+
```
Comment (Shaptic) [inline cmd/soroban-rpc/internal/transactions/transactions_test.go:362]: length check (easier) or content check (harder) would be good here besides not-nil

**Review by Shaptic — APPROVED** [review]
LGTM!

**Comment by psheth9** [comment]
Moving to soroban-rpc

**Pull request closed without merging**